contracts/extensions/BasicIssuanceExtension.sol

```diff
@@ -0,0 +1,248 @@
+    using PreciseUnitMath for uint256;
+    using SafeMath for uint256;
+
+    /* ============ Events ============ */
+
+    event ExtensionInitialized(
+        address _setToken,
+        address _delegatedManager
+    );
+
+    event ExtensionRemoved(
+        address _setToken,
+        address _delegatedManager
+    );
+
+    event FeesDistributed(
+        address _setToken,
+        address indexed _ownerFeeRecipient,
+        address indexed _methodologist,
+        uint256 _ownerTake,
+        uint256 _methodologistTake
+    );
+
+    /* ============ State Variables ============ */
+
+    // Instance of BasicIssuanceModule
+    IIssuanceModule public immutable issuanceModule;
+
+    // Mapping from Set Token to DelegatedManager 
+    mapping(ISetToken => IDelegatedManager) public setManagers;
+
+    /* ============ Constructor ============ */
+
+    constructor(
+        IManagerCore _managerCore,
+        IIssuanceModule _issuanceModule
+    )
+        public
+        BaseGlobalExtension(_managerCore)
+    {
+        issuanceModule = _issuanceModule;
+    }
+
+    /* ============ External Functions ============ */
+
+    /**
+     * ANYONE CALLABLE: Distributes fees accrued to the DelegatedManager. Calculates fees for
+     * owner and methodologist, and sends to owner fee recipient and methodologist respectively.
+     */
+    function distributeFees(ISetToken _setToken) public {
+        IDelegatedManager delegatedManager = _manager(_setToken);
+
+        uint256 totalFees = _setToken.balanceOf(address(delegatedManager));
+
+        address methodologist = delegatedManager.methodologist();
+        address ownerFeeRecipient = delegatedManager.ownerFeeRecipient();
+
+        uint256 ownerTake = totalFees.preciseMul(delegatedManager.ownerFeeSplit());
+        uint256 methodologistTake = totalFees.sub(ownerTake);
+
+        if (ownerTake > 0) {
+            delegatedManager.transferTokens(address(_setToken), ownerFeeRecipient, ownerTake);
+        }
+
+        if (methodologistTake > 0) {
+            delegatedManager.transferTokens(address(_setToken), methodologist, methodologistTake);
+        }
+
+        emit FeesDistributed(address(_setToken), ownerFeeRecipient, methodologist, ownerTake, methodologistTake);
+    }
+
+    /**
+     * ONLY OWNER: Initializes BasicIssuanceExtension to the DelegatedManager.
+     *
+     * @param _delegatedManager     Instance of the DelegatedManager to initialize
+     */
+    function initializeExtension(IDelegatedManager _delegatedManager) external {
+        ISetToken setToken = _delegatedManager.setToken();
+
+        require(
+            managerCore.isFactory(msg.sender) || 
+            address(_delegatedManager) == setToken.manager(),
+            "Must be factory or input must be SetToken manager"
+        );
+        require(msg.sender == _delegatedManager.owner(), "Must be owner");
+        require(_delegatedManager.isPendingExtension(address(this)), "Extension must be pending");
```

Comment:
Since we don't have a way to initialize a module just by itself we should make sure that the module is initialized before this function is called.

Reply:
We should also make a note in the `factory.initialize` function that all modules must be initialized before the extension is initialized

Reply:
https://github.com/SetProtocol/STIPS/pull/16#discussion_r818006251

Reply:
I thought this sequence had a catch-22 you identified last week. 
+ we need to initialize the module using the extension by calling invokeManager
+ invokeManager is gated by `onlyExtension`
+ `onlyExtension` requires that the extension is initialized

Reply:
Leaving `initializeExtension` with no requirement on the module and adding an `initializeModule` function which requires the extension to be initialized.